main.cmake

```diff
@@ -44,6 +45,17 @@ option(MENDER_ARTIFACT_GZIP_COMPRESSION "Enable GZIP compression support when do
 option(MENDER_ARTIFACT_LZMA_COMPRESSION "Enable LZMA compression support when downloading and extracting Artifacts (Default: ON)" ON)
 option(MENDER_ARTIFACT_ZSTD_COMPRESSION "Enable Zstd compression support when downloading and extracting Artifacts (Default: ON)" ON)
 
+# Coverage may fail when building tests.
+if(NOT BUILD_TESTS)
+  find_program(CCACHE ccache)
+  if(CCACHE)
+    message(STATUS "Found ccache at ${CCACHE}")
+    set(CMAKE_C_COMPILER_LAUNCHER ${CCACHE})
+    set(CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE})
+  endif()
+endif()
```

Comment:
I don't think we should enable it by default at all. May I suggest a convenience option instead? Something like:

```
option(CCACHE "Enable ccache" OFF)
```

And then use that to decide.

Btw, I think `CMAKE_LINKER_COMPILER_LAUNCHER` should also be added above.

Reply:
> I don't think we should enable it by default at all. May I suggest a convenience option instead? Something like:
> 
> ```
> option(CCACHE "Enable ccache" OFF)
> ```
> 
> And then use that to decide.
> 
> Btw, I think `CMAKE_LINKER_COMPILER_LAUNCHER` should also be added above.

Good idea. However, that option would override the CCACHE variable from find_program, which seems to make cmake quite upset. So I changed the option to "ENABLE_CCACHE" instead.